cmake/patch_repo.py

```diff
@@ -36,10 +36,20 @@ def main():
         action="store_true",
         help="If a patch in a series cannot be applied, restore the original state instead of leaving patches missing. Return code will be 2 instead of 1.",
     )
+    parser.add_argument(
+        "--3way",
+        action="store_true",
```

Comment:
It looks as if `--3way` was previously unconditionally added, and now won't be added by default, because this new boolean argument defaults to off? Is that on purpose? I don't see anything in the commit message mentioning that it was.

Reply:
The `--3way` option has some side effects that I'm not sure were considered when it was added as the default; namely that it can seemingly apply a patch, but leave conflict markers in place. That will then cause the compile to fail, instead of flagging that there's an issue in patching. The expectation is for a user to deal with the conflicts, so it's less useful in an automated context. Using `--3way` shouldn't be necessary from what I've tried (and indeed the newlib patch command never used it), so I removed it from the default but left the option to enable it in the script.

I'll update the commit message when squashing to mention all this.

Reply:
Fine – that makes sense. If it's disabled on purpose and you say why, I'm happy.